packages/grafana-ui/src/utils/useDelayedSwitch.ts

```diff
@@ -0,0 +1,12 @@
+import { useEffect, useState } from 'react';
+
+type DelayOptions = { min?: number; after?: number };
+export function useDelayedSwitch(value: boolean, { min = 250, after = 250 }: DelayOptions = {}): boolean {
```

Comment:
Can you pls add some comment about what this hook does? It's not obvious then you read the name

Reply:
Yeah will do and any suggestion for the name are welcome, really wasn't sure how to name it :D

Reply:
I can imagine - was thinking about the name for it too, but nothing interesting came to my mind